### ISSUE TITLE
Using action_count or request_size swallows bulk response

### ISSUE THREAD

**Issue by grantr** [github/elastomer-client#92]
Normally a bulk request returns the bulk response:

```
client.bulk do |bulk_request|
  bulk_request.index({"foo" => "bar"}, :id => '1', :index => 'test', :type => 'test')
end

=> {"took"=>61, "errors"=>false, "items"=>[{"index"=>{"_index"=>"test", "_type"=>"test", "_id"=>"1", "_version"=>1, "status"=>201}}]}
```

When using the `:action_count` or `:request_size` options to chunk bulk requests behind the scenes, the response is lost:

```
client.bulk(:action_count => 1) do |bulk_request|
  bulk_request.index({"foo" => "bar"}, :id => '1', :index => 'test', :type => 'test')
  bulk_request.index({"foo" => "bar"}, :id => '2', :index => 'test', :type => 'test')
end

=> nil
```

I think when `:action_count` or `:request_size` are used, the return value should be an array of response hashes.

I can't decide what the return value should be if no requests are made. The non-chunking `bulk` returns nil, but maybe the chunking version should return an empty array.

Is it worth making the multiple-request bulk helper a separate class? We use it heavily, so changing it would be painful, but it has opinions about how to handle multiple requests that maybe don't belong in the single-request bulk helper. Perhaps we could move it to a utility gem and supplement it with other tools like a standard scan/bulk looper.

**Comment by chrismwendt** [comment]
Check out `bulk_stream_items` which was implemented in #112. https://github.com/github/elastomer-client/blob/master/lib/elastomer/client/bulk.rb#L119

**Comment by grantr** [comment]
Fixed by #112.
